student/tests.py

```diff
@@ -105,3 +118,48 @@ def test_get_student_raises_exception_for_non_student_user():
     example_user_not_student.save()
     with pytest.raises(Student.DoesNotExist):
         Student.get_student(example_user_not_student.id)
+
+
+@pytest.mark.django_db
+def test_update_student_account_settings_forms_loads_correctly(example_student_a, example_student_a_data_as_dictionary,
+                                                               client):
+    client.force_login(example_student_a.user)
+    response = client.get("/student/account_settings/")
+    assert response.status_code == 200
+    form = response.context["form"]
+    form_initial_data = response.context["form"].initial
+    assert isinstance(form, UpdateStudentAccountSettingsForm)
+    assert all(form_initial_data[key] == example_student_a_data_as_dictionary[key] for key in form_initial_data)
+
+
+@pytest.mark.django_db
+def test_update_student_account_settings_with_invalid_data(example_student_a, example_student_a_data_as_dictionary,
+                                                           client):
+    invalid_email = "guy"
+    client.force_login(example_student_a.user)
+    example_student_a_data_as_dictionary["email"] = invalid_email
+    response = client.post("/student/account_settings/", data=example_student_a_data_as_dictionary)
+    assert response.status_code == 200
+    form = response.context["form"]
+    assert form.is_valid() is False
+
+
+@pytest.mark.django_db
+def test_update_student_account_settings_with_valid_data(example_student_a, example_student_a_data_as_dictionary,
+                                                         client):
+    new_email = "guyyafe@gmail.com"
+    new_full_name = "guy y"
+    new_about = "I like tests"
+    client.force_login(example_student_a.user)
+    example_student_a_data_as_dictionary["email"] = new_email
+    example_student_a_data_as_dictionary["full_name"] = new_full_name
+    example_student_a_data_as_dictionary["about"] = new_about
+    response = client.post("/student/account_settings/", data=example_student_a_data_as_dictionary)
+    assert response.status_code == 302
```

Comment:
Missing test for the redirection target - there is an HTTP header that contains it in the response.

Reply:
Done!